### PR TITLE
#169469640 - Approving and Rejecting Bookings

### PR DESCRIPTION
**Pull request by nignanthomas**
#### What does this PR do?
Approving and Rejecting Bookings.

#### Description of Task to be completed?
- Add statusId to booking field
- Create route for filtering bookings by status
- Create a patch route to approve/reject bookings
- Create tests for this feature
- Create a route for suppliers and requesters to view a single booking

#### How should this be manually tested?
- Clone the repository from [here](https://github.com/andela/caret-bn-backend.git)
- Browse to the repo directory
- navigate toft-approve-reject-booking-169469640 branch
- Run `npm install`
- run `sequelize db:migrate`
- run `sequelize db:seed:all`
- run `npm run dev`

**To view a single booking**
- In Postman Provide a login bearer token
- send a `GET` request to `http://localhost:<:port>/api/v1/accommodations/bookings/{id}`

**To add a filter bookings**
- In Postman Provide a login bearer token as a supplier
- send a `GET` request to `http://localhost:<:port>/api/v1/accommodations/bookings/search?status={status}`
status can be `pending`, `approved`, `rejected`.

**To approve/reject a booking**
- In Postman Provide a login bearer token as a supplier
- send a `PATCH` request to `http://localhost:<:port>/api/v1/accommodations/bookings/approve/{id}
- send a `PATCH` request to `http://localhost:<:port>/api/v1/accommodations/bookings/reject/{id}

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#169469640](https://www.pivotaltracker.com/story/show/169469640)

#### Screenshots (if appropriate)
Screen Shot 2019-11-07 at 09 41 39
Screen Shot 2019-11-07 at 09 43 09

#### Questions:
N/A

### CODE REVIEW

**Review by ngireric123 — APPROVED** [review]
good job Thomas but you decrease coveralls kindly increase them

**Review by alainmateso — CHANGES_REQUESTED** [review]
Good job Thomas! however, please ensure that a supplier can only see bookings made to accommodations they own and please return data objects when possible. FOr example when viewing bookings, you can return the user's details as well as the booked accommodation details as well as the status name instead of statusId.

**Review by alainmateso — APPROVED** [review]

File: src/controllers/accommodationController.js [hunk 1]
```diff
@@ -218,4 +228,73 @@ export default class AccommodationController {
     }
     return responseUtil(res, 403, strings.users.error.TRAVEL_ADMINS_ONLY);
   }
+
+  static async searchBookings(req, res) {
+    const bookings = await bookingHelper.findBookings(req.body);
+    let myBookings = [];
```
Comment (nakiwuge) [inline src/controllers/accommodationController.js:234]: Add a new line

File: src/database/models/bookingnotifications.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+'use strict';
+module.exports = (sequelize, DataTypes) => {
+  const bookingnotifications = sequelize.define('bookingnotifications', {
```
Comment (nakiwuge) [inline src/database/models/bookingnotifications.js]: change table name  and const to `bookingNotifications`

File: src/tests/bookingsTests.spec.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import { describe, it, before } from 'mocha';
+import app from '../index';
+import mockData from './mockData/mockData';
+
```
Comment (nakiwuge) [inline src/tests/bookingsTests.spec.js:6]: remove this line

File: src/middlewares/bookingFound.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import bookingHelper from '../helpers/bookingHelper';
+import responseError from '../utils/responseError';
+import strings from '../utils/stringsUtil';
+
+const bookingFound = async (req, res, next) => {
+  const { id } = req.params;
+  const booking = await bookingHelper.findOneBooking({ id });
```
Comment (nakiwuge) [inline src/middlewares/bookingFound.js:7]: Add new line

File: src/middlewares/bookingNotPending.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import responseError from '../utils/responseError';
+import strings from '../utils/stringsUtil';
+
+const { BOOKING_APPROVED, BOOKING_REJECTED } = strings.accommodations.success;
+
```
Comment (nakiwuge) [inline src/middlewares/bookingNotPending.js:5]: remove this line

**Review by nakiwuge — CHANGES_REQUESTED** [review]
Good job @nignanthomas, kindly check out the comments I have left on your pr.

**Review by nakiwuge — APPROVED** [review]

**Pull request merged**